### PR TITLE
Redmine PUMI Subversion changes

### PR DESCRIPTION
**Pull request by ibaned**
This pull request will track our effort to merge @seegyoung 's changes from Subversion. It will allow us to do simple code review, etc.

### CODE REVIEW

File: pumi/libpumi.pc.in [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+libdir=@libdir@
+includedir=@includedir@
+
+Name: PUMI
+URL: http://www.scorec.rpi.edu/pumi
+Description:
+Version: @PACKAGE_VERSION@
+Requires: libapf libparma libapf_zoltan libmds
+Libs: -L${libdir} -lpumi
+Cflags: -I${includedir}
```
Comment (cwsmith) [inline pumi/libpumi.pc.in]: @ibaned this can be removed now right?

File: pumi/pumi_ghost.cc [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+    pumi::instance()->ghosted_vec[d].clear();
+  }
+}
+
+// *********************************************************
+void pumi_ghost_getInfo (pMesh, std::vector<int>&)
+// *********************************************************
+{
+  if (!pumi_rank()) 
+    std::cout<<"[PUMI ERROR] "<<__func__<<" failed: not supported\n";
```
Comment (cwsmith) [inline pumi/pumi_ghost.cc]: Should this (and other unsupported functions) be removed until they are available?

File: pumi/pumi_mesh.cc [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+      Parts incoming;
+      apf::unpackParts(incoming);
+      apf::unite(current,incoming);
+      m->setResidence(entity,current);
+    }
+  }
+}
+
+// *********************************************************
+void distribute(pMesh m, Distribution* plan)
```
Comment (cwsmith) [inline pumi/pumi_mesh.cc]: I thought this was  a copy of 'apf::migrate', but this is the multi-layer 'ghosting' for fusion. Can we come up with a less generic name?  @bgranzow suggestions?

Comment (seegyoung) [inline pumi/pumi_mesh.cc]: Distribution is not the same as "multi-layer" ghosting. It allows to support the multiple element copies and each recognizes off-part copies as remote copies.

File: pumi/pumi_ghost.cc (hunk 2 above)
Comment (seegyoung) [inline pumi/pumi_ghost.cc]: Yes, it's your choice.

File: pumi/libpumi.pc.in (hunk 1 above)
Comment (seegyoung) [inline pumi/libpumi.pc.in]: Yes. Please feel free to do whatever necessary for merging.

Comment (ibaned) [inline pumi/libpumi.pc.in]: this is done in commit 76b3320

**Comment by ibaned** [comment]
This is more or less ready to merge at this point. The remaining issues can be addressed post-merge. We're just waiting on coordination with the other merge going on, to make sure we do one at a time carefully.

File: pumi/pumi_mesh.cc (hunk 3 above)
Comment (cwsmith) [inline pumi/pumi_mesh.cc]: I understand there is a functional difference; thanks for the reminder.  How about 'linked ghosts'?

As pointed out though, we can resolve this after the merge.

**Comment by ibaned** [comment]
I just committed a few finishing touches to sync phasta code that had changed, and I'm going to merge this into `develop` now. @seegyoung , please make all future changes to the Github `develop` branch, not in Redmine because it will be harder to merge from Redmine a second time. Also, thank you both for your help in merging.

**Comment by seegyoung** [comment]
What's the URL for develop branch?

**Comment by ibaned** [comment]
same URL, branches are contained within. The easiest thing to do probably is to get your first copy of the code like this:
```
git clone -b develop https://github.com/SCOREC/core.git
```
Then you can use `git push` and `git pull` in a straightforward way.
This stuff is covered in public Git tutorials; I'll try to find a few good ones and compile a list of them.

**Pull request merged**